jme3-core/src/tools/java/jme3tools/optimize/LodGenerator.java

```diff
@@ -576,13 +576,22 @@ public VertexBuffer[] computeLods(TriangleReductionMethod reductionMethod, float
                 numBakedLods++;
             }
         }
-        if (numBakedLods <= lodCount) {
-            VertexBuffer[] bakedLods = new VertexBuffer[numBakedLods];
-            System.arraycopy(lods, 0, bakedLods, 0, numBakedLods);
-            return bakedLods;
-        } else {
-            return lods;
+
+        return cleanBuffer(lods, numBakedLods);
+    }
+
+    private VertexBuffer[] cleanBuffer(VertexBuffer[] lods, int lodLevel) {

```

Comment:
Hi guys, I have a doubt, I have not tried the code so correct me if I am wrong. I have a feeling that this method moves any null values to the end of the array.

 If the old code generated this output `[x, x, null, null, x ,x]`, this new code should produce the following output `[x, x, x, x, null, null]`. 
Is this the purpose of the change?

Reply:
@capdevon  Jup `numBakedLods` is the number of successfully build lods. Maybe changing the lodLevel varibale @riccardobl ?

Reply:
maybe rename it to numComputedLodLevels ?

Reply:
maybe you could do it this way:


```java
/**
 * @param lods ...
 * @param numBakedLods the num of computed lod levels
 */
private VertexBuffer[] cleanBuffer(VertexBuffer[] lods, int numBakedLods) {
	...
}
```

Reply:
Documentation is a bit overkill for a method used once which is located two lines under the usage. BUT the name change makes sense.

Reply:
With the little javadoc I suggested, the changes introduced by your PR would be immediately clear to anyone, without having to read the whole thread or analyze all the complex code in the class, and no one would have any doubts about your solution. I don't have a strong position on this point, since it is a private method, but you would certainly leave important info for those who might come across the analysis of this class again in the future ;)

Reply:
In gerneraly yes. But in this case would be the javadoc as big as the function. You have to read the code anyways, it could be that I've had used the `lodCount ` instead of the `numBakedLods ` and the doc woundn't made anything easier. 
My gut feeling tells me that in this case it would contributes more noise to the code which is as dense as the function itself.  

In addtion I am working on a refactoring of the LodGenerator and currently don't need the cleanBuffer function. Maybe I need this in the futur but it seams unlikly. On the other hand I have the feeling that this refactoring will be hard to get through if I look at the change/comment ratio :)

In the end it comes down to tast. So @riccardobl  javadoc or no javadoc.

Reply:
I don't think there the need to add javadoc, the new variable name is enough. :+1:

Reply:
OK, everything looks fine to me now.  Thank you guys.